### ISSUE TITLE
Kill DeclarationType.ModuleOption

### ISSUE THREAD

**Issue by retailcoder** [rubberduck-vba/Rubberduck#2861]
That one predates version 1.0 I think. Module options aren't *declarations*, and `DeclarationType.ModuleOption` is used in 12 places in the solution:

 - In `API.Declaration` - it should die here too
 - In `Symbols.Declaration` - it's part of the declaration types we exclude as "never array" (duh!)
 - In `Symbols.DeclarationFinder` - where it's special-cased so as to return the parent module when it's selected.
 - In `Symbols.DeclarationSymbolsListener` - which creates the module option "declarations".
 - In `FindSymbolViewModel` - where it's also explicitly excluded from navigatable declarations.
 - In `OptionBaseInspection` - IMO the inspection itself should be removed.
 - In `OptionExplicitInspection` - we should be inspecting the parse tree instead.
 - In `RenameModel.AcquireTarget` - where it's also being excluded; you can't rename that declaration.
 - In `UseMeaningfulNameInspection` - where it's also being excluded, for the same reason as above.

So basically all usages beyond *creating* the declarations are for special-casing the declaration type.

Burn.
